src/swsssdk/dbconnector.py

```diff
@@ -3,10 +3,61 @@
 """
 from . import _connector_map, logger
 from .interface import DBInterface
-
+import os
+import json
 
 # FIXME: Convert to metaclasses when Py2 support is removed. Metaclasses have unique interfaces to Python2/Python3.
 
+class SonicDBConfig(object):
+    SONIC_DB_CONFIG_FILE = "/var/run/redis/sonic-db/database_config.json"
+    _sonic_db_config_init = False
+    _sonic_db_config = {}
+
+    @staticmethod
+    def load_sonic_db_config(sonic_db_file_path=SONIC_DB_CONFIG_FILE):
+        """
+        Get multiple database config from the database_config.json
+        """
+        if SonicDBConfig._sonic_db_config_init == True:
+            return
+
+        try:
+            if os.path.isfile(sonic_db_file_path) == False:
+                sonic_db_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'config', 'database_config.json')
+            with open(sonic_db_file_path, "r") as read_file:
+                SonicDBConfig._sonic_db_config = json.load(read_file)
+        except (OSError, IOError):
+            msg = "Could not open sonic database config file '{}'".format(sonic_db_file_path)
+            logger.exception(msg)
+            raise RuntimeError(msg)
+        SonicDBConfig._sonic_db_config_init = True
+
+    @staticmethod
+    def get_sonic_db_socket(db_name):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        inst_name = SonicDBConfig._sonic_db_config["DATABASES"][db_name]["instance"]
+        return SonicDBConfig._sonic_db_config["INSTANCES"][inst_name]["unix_socket_path"]
+
+    @staticmethod
+    def get_sonic_db_hostname(db_name):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        inst_name = SonicDBConfig._sonic_db_config["DATABASES"][db_name]["instance"]
+        return SonicDBConfig._sonic_db_config["INSTANCES"][inst_name]["hostname"]
+
+    @staticmethod
+    def get_sonic_db_port(db_name):
+        if SonicDBConfig._sonic_db_config_init == False:
+            SonicDBConfig.load_sonic_db_config()
+        inst_name = SonicDBConfig._sonic_db_config["DATABASES"][db_name]["instance"]
+        return SonicDBConfig._sonic_db_config["INSTANCES"][inst_name]["port"]
+
+    @staticmethod
+    def get_sonic_db_id(db_name):
```

Comment:
>get_sonic_db_id [](start = 8, length = 15)

I guess you could rewrite get_dbid instead of creating a new function.

Reply:
done